### ISSUE TITLE
Invalid types using @tsconfig/node20

### ISSUE THREAD

**Issue by javialon26** [openapi-ts/openapi-typescript#1824]
It looks like extending the tsconfig.json file from @tsconfig/node20 is causing type errors, particularly with the second argument (options) on routes and with parameters like params, body, headers, etc., not appearing in VSCode IntelliSense. However, when you remove the lib: ["es2023"] setting, everything works as expected.

Reference #1749 

Packages versions:
"openapi-fetch": "^0.10.5",
"openapi-typescript": "^7.3.0",
"typescript": "^5.5.4"

Generator script:
"generate:client": "openapi-typescript https://petstore3.swagger.io/api/v3/openapi.yaml -o ./src/lib/client.d.ts",

**Comment by javialon26** [comment]
tsconfig.json
```
{
  "extends": "@tsconfig/node20/tsconfig.json",
  "compilerOptions": {
    "outDir": "./dist"
  },
  "include": ["src"]
}
```

Example using pet store swagger file:

<img width="758" alt="Screenshot 2024-08-07 at 2 33 52 AM" src="https://github.com/user-attachments/assets/e7f558db-a356-4f5b-8391-8f4c16ab11ed">

**Comment by ngraef** [comment]
This appears to be broken by a change in `typescript@5.5`, specifically https://github.com/microsoft/TypeScript/pull/58610, which first shipped in `typescript@5.5.0-dev.20240524`. However, the underlying cause is the typing issue noted in #1823.

I can reproduce this issue in a project with:
- `typescript@>=5.5`
- `@types/node@>=18`
- tsconfig.json declares (or extends a config with) a `lib` array that does not include `"DOM"`
